crates/ty_python_semantic/src/types/tuple.rs

```diff
@@ -226,7 +219,6 @@ impl<'db> TupleType<'db> {
         TupleType::new(db, TupleSpec::homogeneous(element))
     }
 
-    #[salsa::tracked]
```

Comment:
caching this method appears to greatly improve this method on `main` (https://github.com/astral-sh/ruff/pull/19840), greatly deteriorate performance on this PR (it necessitates adding slow cycle handling), and greatly speed up things again when applied to https://github.com/astral-sh/ruff/pull/19811, which is stacked on top of this PR